### DISCUSSION TITLE
Request to make Query\Builder::runSelect method public

### DISCUSSION THREAD

**Discussion by daniser** [laravel/framework#34436] [General]
Sometimes I need to lock some rows in the table without fetching them.
The most efficient way to do this now is to call `->count()` on query builder instance:
```
MyModel::where(...)->lockForUpdate()->count();
```
It's much faster than `->get()`, but it'd much better if it'd be possible to call `->runSelect()` directly, but as of now this method is protected.

Is there some elegant workaround for this or is `->count()` maximum I can do?

**Comment by mfn** [comment]
I recently did "open" some other methods for similar reasons, see e.g. https://github.com/laravel/framework/pull/34102

This PR was made before L8 was released. I'm mentioning this because I also made https://github.com/laravel/framework/pull/33953 which eventually was reverted in L7 and then re-applied to L8, which was also still master back then. They reasons due to backward compatibility break, but I didn't realize understand why 🤔
